internal/controllers/machine/machine_controller_phases.go

```diff
@@ -372,3 +362,31 @@ func (r *Reconciler) reconcileCertificateExpiry(ctx context.Context, _ *clusterv
 
 	return ctrl.Result{}, nil
 }
+
+// removeOutdatedOwnerRefs will remove any MachineSet or ControlPlane ownerReferences from Infrastructure Machine and
+// Bootstrap objects referred to by the machine. These ownerReferences are added on  creation as the Machine does not exist
+// when the objects are initially created.
+func removeOutdatedOwnerRefs(cluster *clusterv1.Cluster, m *clusterv1.Machine, obj *unstructured.Unstructured) {
+	cpGVK := getControlPlaneGVKForMachine(cluster, m)
+	for _, owner := range obj.GetOwnerReferences() {
+		if owner.APIVersion == clusterv1.GroupVersion.String() && owner.Kind == "MachineSet" ||
+			cpGVK != nil && owner.APIVersion == cpGVK.GroupVersion().String() && owner.Kind == cpGVK.Kind {
+			ownerRefs := util.RemoveOwnerRef(obj.GetOwnerReferences(), owner)
+			obj.SetOwnerReferences(ownerRefs)
+
+		}
+	}
+}
+
+// getControlPlaneGVKForMachine returns the Kind of the control plane in the Cluster associated with the Machine.
+// This function checks that the machine is managed by a control plane, and then retrieves the kind from the Cluster's
+// .spec.controlPlaneRef.
+func getControlPlaneGVKForMachine(cluster *clusterv1.Cluster, machine *clusterv1.Machine) *schema.GroupVersionKind {
+	if _, ok := machine.GetLabels()[clusterv1.MachineControlPlaneNameLabel]; ok {
```

Comment:
```suggestion
	if _, ok := machine.GetLabels()[clusterv1.MachineControlPlaneLabelName]; ok {
```

nit. Let's use that one, it's slightly more specific to determine if a Machine is a control plane machine (and also exists independent of my PR)

Reply:
In case we only want to implement this for KCP as we only want to do this for KCP. We can also just hard-code dropping KCP ownerRefs.

Reply:
I'm going to resolve this as I think we're agreed on the current validation strategy - but feel free to open again on further review